hieroplatform/src/main/java/org/hiero/sketch/table/VirtualRowSnapshot.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2017 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.hiero.sketch.table;
+
+import org.hiero.sketch.table.api.ITable;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+/**
+ * A pointer to a (projection of a) row in a table (Stored as the row index). The projection is
+ * specified by the schema.
+ */
+public class VirtualRowSnapshot extends BaseRowSnapshot {
+    protected final ITable table;
+    protected final int rowIndex;
+
+    public VirtualRowSnapshot(final ITable table, final int rowIndex) {
+        super(table.getSchema());
+        this.table = table;
+        this.rowIndex = rowIndex;
+    }
+
+    public VirtualRowSnapshot(final ITable table, final int rowIndex, final Schema schema) {
+        super(schema);
+        this.table = table;
+        this.rowIndex = rowIndex;
+    }
+
+    public RowSnapshot materialize() {
+        HashSubSchema hs = new HashSubSchema();
+        this.schema.getColumnNames().forEach(hs::add);
```

Comment:
Maybe you can check whether the schema is the same as the table schema first.
Use ==, for equality of references, not equals().